ocaml-backend/src/main/java/org/kframework/backend/ocaml/DefinitionToOcaml.java

```diff
@@ -1295,20 +1352,29 @@ public void apply(KApply k) {
                     }
                     int arity = getArity(functionLabel);
                     printFunctionParams(sb, arity);
-                    sb.append(" (config: k) (guard: int) : k = let lbl = \n");
+                    sb.append(" (config: k) (guard: int) : k =\n");
+                    indented(sb).append("let lbl = ");
                     encodeStringToIdentifier(sb, functionLabel);
-                    sb.append(" and sort = \n");
+                    sb.append('\n');
+                    indented(sb).append("and sort = ");
                     encodeStringToIdentifier(sb, mainModule.sortFor().apply(functionLabel));
-                    sb.append(" in ");
-                    sb.append("match c with \n");
+                    sb.append(" in\n");
+                    indented(sb).append("try (\n");
```

Comment:
you can't have a try statement here, it will make these functions be unable to become tail recursive, which will break definitions that rely on tail recursion to bound the size of the stack.

Reply:
What if I just added a command-line parameter that would enable this feature only when the user asks for that? This way it would not be a performance problem.

Reply:
That could definitely work, yes. Just bear in mind that this isn't a performance issue, it's a correctness issue; if you try to use this feature on a program that requires tail recursion, your program is going to segfault.

Reply:
Ok. In that case, I will add a test relying on tail recursion.

Reply:
I think you might misunderstand: the stack trace being enabled at kompile time is going to make programs that depend on tail recursion segfault; it doesn't make much sense to add a test that will just fail.

Reply:
I will add a test testing that it will not segfault if the stacktracing is disabled.

Reply:
Ah I see what you're saying. Okay yeah just make a function that recursively calls itself 100000 times.

Reply:
Ok. Btw can't we just eliminate the tail recursion before the backend starts? Just by transforming rules like
```
rule foo(P1::T1, ..., Pn::Tn) => foo(E1, ..., En)
```
to
```
rule foo(P1::T1 => E1, ..., Pn::Tn => En)
```
? Also, I understand the pipeline as
```
input definition => [frontend] => kore definition
                                          v
                                   [transformations]
                                          v
output   <=   [backend]   <=    kore definition
```
; is that correct?